src/core/export.js

```diff
@@ -0,0 +1,255 @@
+var moment = require('moment/moment');
+var _ = require('underscore');
+import XLSX from 'xlsx';
+import * as filesaver from 'filesaver.js';
+import axios from 'axios';
+import urljoin from 'url-join';
+
+/*
+ * Takes current data displayed in the DataTable, Annual Cycle graph, or
+ * Projected Change graph, along with contextual data from user input, creates
+ * an XLSX or CSV file, and serves it to the user for download. Draws on example
+ * code from js-xlsx docs: https://github.com/SheetJS/js-xlsx
+ */
+var exportDataToWorksheet = function(datatype, metadata, data, format, selection) {
+  // create workbook object containing one or more worksheets
+  var wb = {
+      Sheets: {},
+      SheetNames: []
+  };
+
+  // prepare filename, metadata cells, and data cells according to type of export
+  var summaryCells, dataCells, outputFilename;
+  var filenamePrefix = "PCIC_CE_";
+  var filenameInfix = "Export_" + metadata.model_id + "_" + metadata.experiment + "_" + metadata.variable_id;
```

Comment:
Prefer [template literals](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals) to manual concatenation of strings. This would look like:

```
`Export_${metadata.model_id}_${metadata.experiment}_${ metadata.variable_id}`
```